src/components/Calculator.js

```diff
@@ -1,239 +1,236 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import '../index';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      operators: '+-x%÷',
-      numbers: '0123456789',
-      total: '',
-      next: '',
-      operation: '',
-      completed: false,
-      finalObject: {},
-      expression: '',
-    };
-  }
+const Calculator = () => {
+  const operators = '+-x%÷';
+  const numbers = '0123456789';
+  const [total, setTotal] = useState('');
+  const [next, setNext] = useState('');
+  const [operation, setOperation] = useState('');
+  const [expression, setExpression] = useState('0');
+  const [completed, setCompleted] = useState(false);
```

Comment:
- [ ] The hooks are meant to be for calculating not to refactor all, so what you did here you refactored the buttons to be hooked but the requirement says you need to refactor the base class to a function that will take a hook.